source/screens/caseScreens/CaseOverview.js

```diff
@@ -78,19 +79,25 @@ const computeCaseComponent = (status, latestCase, navigation) => {
 const CaseOverview = ({ navigation }) => {
   const [caseItems, setCaseItems] = useState([]);
   const { cases } = useContext(CaseState);
+  const { findFormByType } = useContext(FormContext);
 
   useEffect(() => {
-    const updatedItems = [];
-    caseTypes.forEach(caseType => {
-      const [status, latestCase, relevantCases] = getCaseTypeAndLatestCase(
-        caseType,
-        Object.values(cases)
-      );
-      const component = computeCaseComponent(status, latestCase, navigation);
-      updatedItems.push({ caseType, status, latestCase, component, cases: relevantCases });
-    });
-    setCaseItems(updatedItems);
-  }, [cases, navigation]);
+    const updateItems = async () => {
+      const updateItemsPromises = caseTypes.map(async caseType => {
+        const [status, latestCase, relevantCases] = await getCaseTypeAndLatestCase(
+          caseType,
+          Object.values(cases),
+          findFormByType
```

Comment:
Do we need to send the findFormByType down as a callback? Feels more readable to retrieve the form ids here before we call the getCaseTypeAndLatestCase.